internal/controller/k8s_resources.go

```diff
@@ -99,10 +118,116 @@ func convertMetaObjectToMetadata(logger *zap.SugaredLogger, ctx context.Context,
 			return objMetadata, nil
 		}
 		objMetadata.KindSpecific = &pb.KubernetesObjectData_Node{Node: &pb.KubernetesNodeData{ProviderId: providerId}}
+	case "Service":
+		serviceAttributes, err := getServiceAttributes(ctx, obj.GetName(), clientset, obj.GetNamespace())
+		if err != nil {
+			return objMetadata, nil
+		}
+		convertedServiceData := convertToKubernetesServiceData(serviceAttributes)
+		objMetadata.KindSpecific = &pb.KubernetesObjectData_Service{Service: convertedServiceData}
+
 	}
 	return objMetadata, nil
 }
 
+func convertIngressToStringList(ingress []v1.LoadBalancerIngress) []string {
+	result := []string{}
+	for _, i := range ingress {
+		if i.IP != "" {
+			result = append(result, i.IP)
+		}
+		if i.Hostname != "" {
+			result = append(result, i.Hostname)
+		}
+	}
+	return result
+}
+
+func convertServicePortsToNodePorts(servicePorts []v1.ServicePort) []*Ports {
+	nodePorts := make([]*Ports, 0, len(servicePorts))
+	for _, sp := range servicePorts {
+		if sp.NodePort != 0 {
+			nodePort := &Ports{
+				NodePort: sp.NodePort,
+				Port:     sp.Port,
+				Protocol: string(sp.Protocol),
+			}
+			nodePorts = append(nodePorts, nodePort)
+		}
+	}
+	return nodePorts
+}
+
+func getServiceAttributes(ctx context.Context, serviceName string, clientset *kubernetes.Clientset, namespace string) (*ServiceAttributes, error) {
+	service, err := clientset.CoreV1().Services(namespace).Get(ctx, serviceName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.New("failed to get service")
+	}
+	loadBalancerIngress := []string{}
+	if len(service.Status.LoadBalancer.Ingress) > 0 {
+		loadBalancerIngress = convertIngressToStringList(service.Status.LoadBalancer.Ingress)
+	}
+	loadBalancerClass := ""
+	if service.Spec.LoadBalancerClass != nil {
+		loadBalancerClass = *service.Spec.LoadBalancerClass
+	}
+	attributes := &ServiceAttributes{
+		ClusterIPs:          service.Spec.ClusterIPs,
+		ExternalIPs:         service.Spec.ExternalIPs,
+		LoadBalancerIngress: loadBalancerIngress,
+		LoadBalancerIP:      service.Spec.LoadBalancerIP,
+		LoadBalancerClass:   loadBalancerClass,
+	}
+
+	if service.Spec.Type == v1.ServiceTypeNodePort || service.Spec.Type == v1.ServiceTypeLoadBalancer {
+		attributes.NodePorts = convertServicePortsToNodePorts(service.Spec.Ports)
+	}
```

Comment:
Convert ports unconditionally regardless of the service type.
```suggestion
	attributes.Ports = convertServicePortsToNodePorts(service.Spec.Ports)
```